### PR TITLE
Wipe owner info as soon as possible

### PR DESCRIPTION
**Pull request by sttts**
We don't need the user info on a) ready workspaces b) on user-visible Workspaces.

### CODE REVIEW

File: pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile_metadata.go [hunk 1]
```diff
@@ -58,6 +58,16 @@ func (r *metaDataReconciler) reconcile(ctx context.Context, workspace *tenancyv1
...
 	}
 
+	if workspace.Status.Phase == tenancyv1alpha1.ClusterWorkspacePhaseReady {
+		if _, found := workspace.Annotations[tenancyv1alpha1.ClusterWorkspaceOwnerAnnotationKey]; found && len(workspace.Annotations) == 1 {
+			workspace.Annotations = nil
+			changed = true
+		} else if found {
+			delete(workspace.Annotations, tenancyv1alpha1.ClusterWorkspaceOwnerAnnotationKey)
+			changed = true
+		}
```
Comment (s-urbaniak) [inline pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile_metadata.go:68]: just a nit, not sure if this reads better:
```
		if _, found := workspace.Annotations[tenancyv1alpha1.ClusterWorkspaceOwnerAnnotationKey]; found {
			delete(workspace.Annotations, tenancyv1alpha1.ClusterWorkspaceOwnerAnnotationKey)
			changed = true
		}

		if len(workspace.Annotations) == 0 {
			workspace.Annotations = nil
			changed = true
		}
```

Comment (s-urbaniak) [reply pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile_metadata.go:68]: please gdoc why setting the annotation map to nil is necessary.

Comment (sttts) [reply pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile_metadata.go:68]: it's not necessary actually. Removed.

**Pull request merged**